go/vm/geth/geth.go

```diff
@@ -117,7 +126,9 @@ func createGethInterpreterContext(parameters vm.Parameters) (*geth.EVM, *geth.Co
 
 	// Create empty block context based on block number
 	blockCtx := geth.BlockContext{
-		BlockNumber: big.NewInt(int64(parameters.Revision)*10 + 2),
+		// Note: after Paris,the block number no longer indicates the revision
+		// but the existence of the random field in the block context.
```

Comment:
I think this comment can be removed. The `BlockNumber` in the context serves its independent purpose (to provide the input for the `NUMBER` op-code), and I do not think the expectation is that this field is influencing the revision as this is handled by other code in this function.

Reply:
So far this was the only place forwarding any information mapping our revision enum into geth.
(only use of parameters. Revision in the function).
Nevertheless, the new comment reflects the intent better.